src/Lean/Elab/Quotation/Precheck.lean

```diff
@@ -141,4 +141,36 @@ private def isSectionVariable (e : Expr) : TermElabM Bool := do
   runPrecheck singleQuot.getQuotContent
   adaptExpander (fun _ => pure singleQuot) stx expectedType?
 
+section ExpressionTree
+
+@[builtin_quot_precheck Lean.Parser.Term.binrel] def precheckBinrel : Precheck
+  | `(binrel% $_ $a $b) => do precheck a; precheck b
```

Comment:
What about the first argument, shouldn't we precheck it as well for the sake of completeness?

Reply:
Yes, that makes sense. Done.